Simplify model integration, add streaming, and add MR model support

zephyrie opened the pull request:
Summary
- Add multi-model support for NV-Segment (CT and CTMR variants) and NV-Generate (CT and MR)
- Improve model configuration with centralized paths and fail-fast validation

Changes
- NV-Segment: Support both NV-Segment-CT (132 classes) and NV-Segment-CTMR (345+ classes)
- NV-Generate: Separate CT and MR into distinct directories, use diff_model_infer workflow
- NV-Reason: Include streaming inference with better error handling
- UI improvements: Model-specific recommended settings, lower GPU-friendly defaults
- Documentation: Update README with corrected download instructions and directory structure

zephyrie committed:
Simplify model integration, add streaming, and add MR model support

andinet commented:
@zephyrie  Thank you for this great contribution. I pulled your PR, built and tested it on my machine

1) NV-Segment works well for both CT and CTMR. Excellent! Thanks

2) NV-Generate. I encountered a 'no module available' error for scripts.diff_model_infer. see below. I haven't had a chance to dig into it yet. This happens for both CT and MRI generation

====================
Received NV-Generate-MR generation request with ID: nv-generate-mr-1763212700486
Modality: MR, Version: rflow-mr
Parameters: {'output_size': [128, 128, 128], 'spacing': [1.5, 1.5, 1.5], 'autoencoder_sliding_window_infer_size': [48, 48, 48], 'autoencoder_sliding_window_infer_overlap': 0.5, 'num_splits': 16}
RPC generateWithNVGenerate raised an exception
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.12/concurrent/futures/process.py", line 264, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/local/KHQ/andinet.enquobahrie/s/volview-gtc2025-demo/server/2025_nvidiagtcdc/nv_generate.py", line 162, in do_maisi_generation
    from scripts.diff_model_infer import diff_model_infer
ModuleNotFoundError: No module named 'scripts.diff_model_infer'
"""

===============

3) Reasoning

Streaming was interruped error

<img width="2244" height="1194" alt="image" src="https://github.com/user-attachments/assets/73097837-1748-4e6a-b553-5b0c357d8aef" />

I wonder if there is a data conversation issue. Here is from the console 

NVReasonCXRModule.vue:228 Stream error (handled): Error: /work/ITK-source/ITK/Modules/Core/Common/src/itkMetaDataDictionary.cxx:77:
ITK ERROR: Key 'pointData' does not exist 
    at RpcClient.onStreamResultEvent (client.ts:349:23)


Before testing all the new functionalities, I ran the model health installation verification scripts that you provided, and everything seems to be installed correctly. If I find time to investigate further these issues, I will share my findings.

Commit:
Fix chat streaming by disabling auto-deserialization for image-cache store

The RPC layer's default_deserializers automatically convert vtkjs data to ITK images,
causing getVtkImageData() to return ITK images instead of vtkjs dictionaries.

Commit 38ebdac9 added convert_vtkjs_rgb_to_grayscale() which expected a vtkjs dict
but received an ITK image, causing chat streaming to fail silently.

Solution: Disable auto-deserialization by passing transform_args=False when accessing
the image-cache store. This makes getVtkImageData() return actual vtkjs data as the
name suggests, and each endpoint explicitly converts to ITK as needed.

Changes:
- chat.py: Add transform_args=False, manually convert vtkjs to ITK
- nv_segment.py: Add transform_args=False (existing conversion now works correctly)
- example_api.py: Add transform_args=False, manually convert vtkjs to ITK

This minimal change fixes the streaming bug while making the API behavior match its name.

andinet commented:
@zephyrie  

I pushed a fix to the streaming_plus_mr_models branch to resolve the chat streaming issue (commit 860805e).
The problem was caused by the RPC layer auto-deserializing vtkjs data into ITK images. The fix disables auto-deserialization so getVtkImageData() returns the expected vtkjs dictionary.

Could you test it on your side and let me know if everything works for you?

On my end, it looks good

<img width="2451" height="1339" alt="image" src="https://github.com/user-attachments/assets/29cb103f-cb82-4322-88e8-fb407711c411" />

Commit:
Fix NV-Generate import error in subprocess

- Add ProcessPoolExecutor initializer to configure worker sys.path
- Remove server path from sys.path to avoid scripts/ namespace conflict
- Clear cached scripts module to force fresh import from bundle
- Resolves 'ModuleNotFoundError: No module named scripts.diff_model_infer'

The issue was that /server/scripts and /models/NV-Generate-*/scripts
created a Python namespace package, preventing correct module import.
The worker initializer ensures each subprocess has the correct paths.

zephyrie committed:
Fix NV-Reason streaming and image handling

- Fix vtk_to_itk_image to handle JSON-serialized list data (not just bytes)
- Fix vtk_to_itk_image to handle multi-component images (RGB) by extracting first component
- Clear chat history when user switches images in NVReasonCXRModule.vue

andinet commented:
LGTM!
